suite-native/graph/src/components/PriceChangeIndicator.tsx

```diff
@@ -1,92 +1,35 @@
-    hasIncreased ? 'textPrimaryDefault' : 'textAlertRed';
-
 type PercentageChangeAtom = Atom<number>;
 type HasPriceIncreasedAtom = Atom<boolean>;
 
-type PercentageChangeProps = {
-    percentageChangeAtom: PercentageChangeAtom;
-    hasPriceIncreasedAtom: HasPriceIncreasedAtom;
-};
-
-const PercentageChange = ({
-    percentageChangeAtom,
-    hasPriceIncreasedAtom,
-}: PercentageChangeProps) => {
-    const [percentageChange] = useAtom(percentageChangeAtom);
-    const [hasPriceIncreased] = useAtom(hasPriceIncreasedAtom);
-
-    return (
-        <Text color={getColorForPercentageChange(hasPriceIncreased)} variant="hint">
-            {percentageChange.toFixed(2)}%
-        </Text>
-    );
-};
-
-const PercentageChangeArrow = ({
-    hasPriceIncreasedAtom,
-}: {
-    hasPriceIncreasedAtom: HasPriceIncreasedAtom;
-}) => {
-    const [hasPriceIncreased] = useAtom(hasPriceIncreasedAtom);
-
-    const iconName: IconName = hasPriceIncreased ? 'arrowUp' : 'arrowDown';
-
-    return (
-        <Icon
-            name={iconName}
-            color={getColorForPercentageChange(hasPriceIncreased)}
-            size="extraSmall"
-        />
-    );
-};
-
-const arrowStyle = prepareNativeStyle(() => ({
-    marginRight: 4,
-}));
-
-const priceIncreaseWrapperStyle = prepareNativeStyle<{ hasPriceIncreased: boolean }>(
-    (utils, { hasPriceIncreased }) => ({
-        backgroundColor: hasPriceIncreased
-            ? utils.colors.backgroundPrimarySubtleOnElevation0
-            : utils.colors.backgroundAlertRedSubtleOnElevation0,
-        flexDirection: 'row',
-        alignItems: 'center',
-        paddingHorizontal: utils.spacings.small,
-        paddingVertical: utils.spacings.small / 4,
-        borderRadius: utils.borders.radii.round,
-    }),
-);
-
 export const PriceChangeIndicator = ({
     hasPriceIncreasedAtom,
     percentageChangeAtom,
 }: PriceChangeIndicatorProps) => {
-    const { applyStyle } = useNativeStyles();
-    const [hasPriceIncreased] = useAtom(hasPriceIncreasedAtom);
+    const percentageChange = useAtomValue(percentageChangeAtom);
+    const hasPriceIncreased = useAtomValue(hasPriceIncreasedAtom);
+
+    const icon: IconName = hasPriceIncreased ? 'arrowUp' : 'arrowDown';
+    const badgeVariant: BadgeVariant = hasPriceIncreased ? 'green' : 'red';
+    const percentageChangeLabel = `${percentageChange.toFixed(2)}%`;
 
     return (
-        <Box style={applyStyle(priceIncreaseWrapperStyle, { hasPriceIncreased })}>
-            <Box style={applyStyle(arrowStyle)}>
-                <PercentageChangeArrow hasPriceIncreasedAtom={hasPriceIncreasedAtom} />
-            </Box>
-            <PercentageChange
-                hasPriceIncreasedAtom={hasPriceIncreasedAtom}
-                percentageChangeAtom={percentageChangeAtom}
-            />
-        </Box>
+        <Badge
+            icon={icon}
+            iconSize="extraSmall"
+            variant={badgeVariant}
+            label={percentageChangeLabel}
+        />
```

Comment:
All right. I wasn't aware of this. I will check the performance of this implementation.

Reply:
I agree with Bohdan on this, this will add significant amount of work and this component could be rerendered hundrets times per second if you swipe fast enough through graph. That's reason why value must be separated to smallest possible component aka text only or something.

Reply:
reverted in 0370f3e50. You were right, the new implementation was rerendering about 10x faster than the original one 😅.

Reply:
10x faster? You mean slower, no? @PeKne